charts/garden-project/charts/project-rbac/templates/clusterrole-project-admin.yaml

```diff
@@ -0,0 +1,24 @@
+---
+apiVersion: {{ include "rbacversion" . }}
+kind: ClusterRole
+metadata:
+  name: garden.sapcloud.io:system:project:{{ .Values.project.name }}
+  ownerReferences:
+  - apiVersion: garden.sapcloud.io/v1beta1
+    kind: Project
+    blockOwnerDeletion: false
```

Comment:
Consider rare race condition. If the project is successfully deleted and in but garbage collector has not yet  deleted clusterrole or binding. And somebody created project with same name. For short time old project owner will have access to new project as well.

Reply:
How is that? When the new project with the same name get's created the existing ClusterRoleBinding (if not yet picked up by the garbage collector) would be overridden with the new subjects.

Reply:
Ohh yeah. how could i miss that! 

But with this argument, it would actually make more sense to make it `true` for different reason. Now, let's say garbage collector just applied `deletionTimestamp` on it and k8s-controller came down. The gardener-controller-manager will reconcile it with new owner for new project but obviously `deletionTimestamp` will still be there. So as soon as k8s-controller comes up, it will delete clusterrole and binding. So, it will take another project controller reconciliation to get back new clusterrole and binding.

Reply:
> Now, let's say garbage collector just applied deletionTimestamp on it and k8s-controller came down. 

What do you mean? There is no `finalizer` on `ClusterRole(Binding)`s, hence, as soon as the garbage collector has applied the `deletionTimestamp` the object will be deleted from etcd.